src/Autocomplete/src/Form/AutocompleteEntityTypeSubscriber.php

```diff
@@ -73,11 +73,15 @@ public function preSubmit(FormEvent $event)
                     ++$idx;
                 }
 
-                $options['choices'] = $repository->createQueryBuilder('o')
-                    ->where(sprintf("o.$idField IN (%s)", implode(', ', array_keys($params))))
-                    ->setParameters(new ArrayCollection($params))
-                    ->getQuery()
-                    ->getResult();
+                $queryBuilder = $repository->createQueryBuilder('o');
+
+                if ($params) {
+                    $queryBuilder
+                        ->where(sprintf("o.$idField IN (%s)", implode(', ', array_keys($params))))
+                        ->setParameters(new ArrayCollection($params));
```

Comment:
My two cents, using `->setParameters()` will override the previous bound parameters (if any) to the `QueryBuilder`.

IMO it would be preferable to use `->setParameter()` for each `$params`, to be future-proof.

Reply:
There can't be any previous parameters, because the query is constructed 4 lines before :)

Reply:
For the moment yes, but we can't predict the future.

Reply:
Imho they could be added to $params,no? :)